pkg/reconciler/ingress/ingress_test.go

```diff
@@ -272,10 +272,14 @@ func withPreferHeaderPaths(isAlwaysAsync bool) ingressCreationOption {
 		for _, rule := range ing.Spec.Rules {
 			newRule := rule
 			newPaths := make([]v1alpha1.HTTPIngressPath, 0)
+			appendHeaders := make(map[string]string)
+			appendHeaders["Async-Original-Host"] = getClusterLocalDomain(ingName, defaultNamespace)
 			if isAlwaysAsync {
 				for _, path := range rule.HTTP.Paths {
 					defaultPath := path
 					defaultPath.Splits = splits
+					defaultPath.RewriteHost = getClusterLocalDomain(producerServiceName, "knative-testing")
```

Comment:
this loop seems a little duplicative of the actual logic under test.. we're not _really_ testing if the test and the code are copy/pastes of each other.. wonder if it's worth having the test be more explicit about what it wants here

Reply:
I hear you -- I was trying to confirm that when we provide the initial ingress that our created ingress is as we expect it to be. If the logic is the same, all we're really testing though is that it's created. 🤔 I think i agree, if we just explicitly define what the ingress is rather than using the logic, that may be more reasonable.

Reply:
@julz okay, i explicitly defined the ingresses -- let me know if that works.